bssl-compat/boringssl.cmake

```diff
@@ -0,0 +1,21 @@
+include(ExternalProject)
+
+ExternalProject_Add(
+  BoringSSL
+
+  DOWNLOAD_EXTRACT_TIMESTAMP TRUE
+  URL "https://github.com/google/boringssl/archive/${BORINGSSL_COMMIT}.tar.gz"
```

Comment:
You can't download anything in the Bazel build sandbox. Does the build work also without the BoringSSL download? BoringSSL headers are installed in the include path when building under Envoy Bazel.

Reply:
For some reason, bazel is using the `processwrapper-sandbox` on my machine, which does allow downloads, so I didn't spot that!

As it stands on this branch, the bssl-compat build will not work without the BoringSSL download. The point of this change, is to use files from the actual downloaded BoringSSL rather than have the files duplicated by checking them in.

Reply:
Could we stash a local copy of the BoringSSL header files in a suitable location while keeping the rest of the modifications? The downloading part pins the BoringSSL (API) version to a particular release anyway, and needs a new commit when it is updated. Same happens when updating BoringSSL header files from a new release version.

Reply:
I have replaced the downloading of BoringSSL with a git submodule (at `bssl-compat/boringssl`).